guides/leon/vsftpd.md

```diff
@@ -0,0 +1,211 @@
+# VSFTP GUIDE
+
+## Summary 
+This is the setup guide for a vsftp box
+
+### Contact
+- Twitter: @Dubliyu
+- Slack: @yourlocalgod on wcscusf.slack.com
+- Email: [Gmail](mailto:cleonromero@mgail.com)
+
+### Table
+1. [Prerequisites](#pre-reqs)
+2. [Setup Ubuntu VM](#setup-vm)
+3. [Install VSFTPD](#install)
+4. [Secure VSFTPD](#secure)
+5. [Users](#users)
+6. [Testing](#testing)
+7. [Useful Commands](#useful)
+
+<a id="pre-reqs"></a>
+## Prerequisites 
+1. Have your [virtual environment](https://silexone.github.io/guides/nestor/ISPsetup.html) configured
+2. Have the [ISP](https://silexone.github.io/guides/nestor/ISPsetup.html) gateway running.
+3. Have pfSense running.
+
+
+<br>
+
+<a id="setup-vm"></a>
+## Setup Ubuntu VM 
+1. First, go get the Debian server from ISO [here](https://www.ubuntu.com/download/server/thank-you?version=16.04.3&architecture=amd64). 
+
+    Then, open up VirtualBox, create a new Linux Ubuntu (64-bit) VM, the default setting will do. Then alter the network settings to use **Host-only** adapter instead of NAT. Insert the downloaded ISO into the virtual optical drive and boot.
+   
+2. Select default options during the installation. 
+3. Now you should see a terminal login prompt. Login.
+![1](/blog/content/images/2018/02/1.PNG)
+<br>
+
+<a id="install"></a>
+## Install vsftpd 
+Here we will install and configure vsftpd. Some background knowledge, FTP servers can be either passive or active. If active, the client connects to a random port and port 21 serves as a control port - this creates some problems with firewalls btw. If passive, connect first to port 21 then when requesting a file the transfer will move onto a random port. Ours is passive by default.
+
+1. Preparations
+
+    First become superuser and fetch updates
+    ``` bash
+    sudo su
+    apt-get update
+    apt-get upgrade
+    ```
+2. Install vsftpd
+    
+    ``` bash
+    apt-get install -y vsftpd
+    ```
+3. Enable the vsftpd service to run on boot.
+
+    ```bash
+    systemctl start vsftpd
+    systemctl enable vsftpd
+    ```
+4. Verify that VSFTPD is listening on port 21
+    
+    Run `netstat -ant`
+     ```bash
+     tcp6   0    0   :::21         :::*         LISTEN
+     ```
+5. Now go test it out, [here](#test-unsecure)
+<br>
+
+<a id="secure"></a>
+## Secure vsftpd
+Here we will configure and secure our vsftpd server. Protip: on the ubuntu machine the conf files is in `/etc/vsftpd.conf` but in non-debian distros its generally inside `/etc/vsftpd/vsftpd.conf`.
+ 
+1.  First generate some keys

```

Comment:
You might need to explain or link to a guide/video on public keys so thet understand why they're doing it.